src/component/@common/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  createContext,
+  ReactNode,
+  PropsWithChildren,
+  ChangeEvent,
+  SelectHTMLAttributes,
+  OptionHTMLAttributes,
+} from 'react';
+import {
+  dropdownArrow,
+  dropdownContainer,
+  dropdownOption,
+  dropdownSelect,
+} from './Dropdown.styles';
+import { CardBrand, useCard } from '../../../context/CardContext';
+
+interface DropdownContextType {
+  selectedOption: string;
+  handleChange: (event: ChangeEvent<HTMLSelectElement>) => void;
+}
+
+interface DropdownProps extends SelectHTMLAttributes<HTMLSelectElement> {
+  defaultValue?: string;
+  placeholder?: string;
+  onNext?: () => void;
+}
+
+interface DropdownOptionProps extends OptionHTMLAttributes<HTMLOptionElement> {
+  value: string;
+  children: ReactNode;
+}
+
+export const DropdownContext = createContext<DropdownContextType>({
+  selectedOption: '',
+  handleChange: () => {},
+});
+
+const Dropdown = ({
+  children,
+  defaultValue = '',
+  placeholder = '카드사를 선택해주세요',
+  onNext,
+  ...props
+}: PropsWithChildren<DropdownProps>) => {
+  const { selectedCardBrand, setSelectedCardBrand } = useCard();
+
+  const handleChange = (event: ChangeEvent<HTMLSelectElement>) => {
+    const value = event.target.value as CardBrand;
+    setSelectedCardBrand(value);
+  };
+
+  if (selectedCardBrand) {
+    onNext?.();
+  }
```

Comment:
`onNext`가 Dropdown을 그리는 과정 에서 사이드이펙트처럼 호출되고 있네요.
아래처럼 이벤트 핸들러에서 직접 호출하는 것과 어떤 차이가 있을까요?

```suggestion
  const handleChange = (event: ChangeEvent<HTMLSelectElement>) => {
    const value = event.target.value as CardBrand;
    setSelectedCardBrand(value);
    onNext?.();
  };
```

Reply:
와 제가 왜 전자처럼 코드를 작성했을까요?
의미없는 if문인 것 같아요... 꼼꼼한 리뷰 감사합니다!

465539b40f49fbd916f294940ae75b378f88de53